app/forms/steps/client/has_partner_form.rb

```diff
@@ -1,49 +1,46 @@
 module Steps
   module Client
     class HasPartnerForm < Steps::BaseFormObject
-      attribute :client_has_partner, :value_object, source: YesNoAnswer
+      include Steps::HasOneAssociation
+      has_one_association :partner_detail
 
-      validates_inclusion_of :client_has_partner, in: :choices
+      attribute :has_partner, :value_object, source: YesNoAnswer
+
+      validates_inclusion_of :has_partner, in: :choices
 
       def choices
         YesNoAnswer.values
       end
 
-      private
-
       def changed?
-        !crime_application.client_has_partner.eql?(client_has_partner.to_s)
+        !partner_detail.has_partner.eql?(has_partner.to_s)
       end
 
-      # TODO: When FeatureFlag.partner_journey is removed,
-      # move this form into /partner and Partner module
-      def persist!
+      private
+
+      def persist! # rubocop:disable Metrics/MethodLength
         return true unless changed?
 
         ::CrimeApplication.transaction do
-          reset!
-
-          crime_application.update!(
-            attributes
-          )
+          if has_partner.no?
+            reset!
+          else
```

Comment:
Do we want to explicit here and check that has_partner is yes as someone can click save and come back

Reply:
@mseedat-moj i think it's better to be explicit here

Reply:
Ah never mind, line 22 covers it